### ISSUE TITLE
Poloniex minimum amount

### ISSUE THREAD

**Issue by MixMe** [ccxt/ccxt#2706]
![firefox_screenshot_2018-04-30t10-44-27 922z](https://user-images.githubusercontent.com/3535911/39423881-b2063ff0-4c7c-11e8-869f-2d206029e89f.png)

but i got: 

ccxt\ExchangeError
poloniex => unknown $error => {"error":"Amount must be at least 0.000001."}   

minimum amount is 0.000001. not  0.00000001(1.0E-8)

    OS: Ubuntu Server 16.04
    Programming Language version: PHP
    CCXT version: 1.13.54
    Exchange: poloniex
    Method: load_markets

**Comment by kroitor** [comment]
Can you please show the verbose output of the request as described here: 

https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#how-to-submit-an-issue

?

```
$exchange->verbose = true;
$exchange->createOrder ...
```

What does it print out?

**Comment by MixMe** [comment]
Request: Array ( [0] => GET [1] => https://poloniex.com/public?command=returnTicker [2] => Array ( ) [3] => ) Response: Array ( [0] => GET [1] => https://poloniex.com/public?command=returnTicker [2] => 200 [3] => [4] => Array ( [date] => Array ( [0] => Tue, 01 May 2018 08:41:32 GMT ) [content-type] => Array ( [0] => application/json ) [transfer-encoding] => Array ( [0] => chunked ) [connection] => Array ( [0] => keep-alive ) [set-cookie] => Array ( [0] => __cfduid=d760e756020a9cc2505a1ee629321787c1525164092; expires=Wed, 01-May-19 08:41:32 GMT; path=/; domain=.poloniex.com; HttpOnly ) [access-control-allow-origin] => Array ( [0] => * ) [access-control-allow-methods] => Array ( [0] => GET ) [content-encoding] => Array ( [0] => gzip ) [expect-ct] => Array ( [0] => max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct" ) [server] => Array ( [0] => cloudflare ) [cf-ray] => Array ( [0] => 4140f317aa75b6e6-KIV ) ) [5] => {"BTC_BCN":{"id":7,"last":"0.00000058","lowestAsk":"0.00000059","highestBid":"0.00000058","percentChange":"-0.01694915","baseVolume":"60.62768345","quoteVolume":"102868436.86762334","isFrozen":"0","high24hr":"0.00000062","low24hr":"0.00000057"},"BTC_BELA":{"id":8,"last":"0.00001424","lowestAsk":"0.00001425","highestBid":"0.00001409","percentChange":"0.01714285","baseVolume":"2.06018185","quoteVolume":"146634.53117871","isFrozen":"0","high24hr":"0.00001457","low24hr":"0.00001372"},"BTC_BLK":{"id":10,"last":"0.00003435","lowestAsk":"0.00003445","highestBid":"0.00003412","percentChange":"0.03029394","baseVolume":"3.88025777","quoteVolume":"114450.60781989","isFrozen":"0","high24hr":"0.00003520","low24hr":"0.00003287"},"BTC_BTCD":{"id":12,"last":"0.01406500","lowestAsk":"0.01406502","highestBid":"0.01406500","percentChange":"0.30222076","baseVolume":"15.20287190","quoteVolume":"1196.45303945","isFrozen":"0","high24hr":"0.01420000","low24hr":"0.01068138"},"BTC_BTM":{"id":13,"last":"0.00007622","lowestAsk":"0.00007716","highestBid":"0.00007624","percentChange":"0.02034805","baseVolume":"0.61524989","quoteVolume":"8188.28302181","isFrozen":"0","high24hr":"0.00007781","low24hr":"0.00007287"},"BTC_BTS":{"id":14,"last":"0.00004157","lowestAsk":"0.00004153","highestBid":"0.00004135","percentChange":"0.15987723","baseVolume":"1474.51732577","quoteVolume":"36908489.88540621","isFrozen":"0","high24hr":"0.00004340","low24hr":"0.00003555"},"BTC_BURST":{"id":15,"last":"0.00000269","lowestAsk":"0.00000269","highestBid":"0.00000267","percentChange":"0.01127819","baseVolume":"29.19836353","quoteVolume":"10883677.09339089","isFrozen":"0","high24hr":"0.00000270","low24hr":"0.00000262"},"BTC_CLAM":{"id":20,"last":"0.00055302","lowestAsk":"0.00055662","highestBid":"0.00055302","percentChange":"-0.03562647","baseVolume":"7.84693352","quoteVolume":"14026.66921094","isFrozen":"0","high24hr":"0.00058033","low24hr":"0.00054000"},"BTC_DASH":{"id":24,"last":"0.05133010","lowestAsk":"0.05156915","highestBid":"0.05133010","percentChange":"-
......

**Comment by MixMe** [comment]
at the and:

 Request: Array ( [0] => POST [1] => https://poloniex.com/tradingApi [2] => Array ( [0] => Content-Type: application/x-www-form-urlencoded [1] => Key: ******* [2] => Sign: ******* ) [3] => command=buy¤cyPair=USDT_ETH&rate=0.00000001&amount=1.0E-8&fillOrKill=1&nonce=1525164417197 ) Response: Array ( [0] => POST [1] => https://poloniex.com/tradingApi [2] => 422 [3] => [4] => Array ( [date] => Array ( [0] => Tue, 01 May 2018 08:46:59 GMT ) [content-type] => Array ( [0] => application/json ) [transfer-encoding] => Array ( [0] => chunked ) [connection] => Array ( [0] => keep-alive ) [set-cookie] => Array ( [0] => __cfduid=db0d9406a21d4b93cc438d116572a853e1525164418; expires=Wed, 01-May-19 08:46:58 GMT; path=/; domain=.poloniex.com; HttpOnly ) [cache-control] => Array ( [0] => private ) [expect-ct] => Array ( [0] => max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct" ) [server] => Array ( [0] => cloudflare ) [cf-ray] => Array ( [0] => 4140fb121a8cb6d4-KIV ) ) [5] => {"error":"Amount must be at least 0.000001."} )

**Comment by ccasselman** [comment]
For the record, I just verified if I call the their API directly with the same values it goes through successfully.

**Comment by ccasselman** [comment]
Above and in mine the call looks like `command=buy¤cyPair=USDT_ETH&rate=0.00000001&amount=1.0E-8&fillOrKill=1&nonce=1525164417197`

When it should be `command=buyPair=USDT_ETH&rate=0.00000001&amount=1.0E-8&fillOrKill=1&nonce=1525164417197`

I think `¤cy` is getting inserted by accident?

**Comment by kroitor** [comment]
@ccasselman which particular params do you use with `createOrder` call? Can you please follow these recommendations precisely: https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#how-to-submit-an-issue? We need to see the code to reproduce it (without your keys) + verbose request and response.

The `¤cy` thing is just how it's rendered on a webpage. It's actually `&currencyPair`, but stupid parsers render &curr as ¤.

**Comment by ccasselman** [comment]
I see about the `&curr`.

Here is my call:
```
        $exchangeName = 'poloniex';
        $exchange_class = '\\ccxt\\' . strtolower($exchangeName);
        $client = new $exchange_class(
            [
                'apiKey' => *****,
                'secret' => *****,,
                'verbose' => true
            ]
        );
$info = $client->create_order($pair, Constants::LIMIT, $type, $price, $quantity);
```
I posted the request and response in #2903.

**Comment by kroitor** [comment]
```
$info = $client->create_order($pair, Constants::LIMIT, $type, $price, $quantity);
```

You misplaced price and quantity. Quantity comes first: https://github.com/ccxt/ccxt/wiki/Manual#placing-orders

The problem with these messages

```
Amount must be at least 0.000001.
```

is due to the poor Poloniex API and docs – they don't serve proper precision information from their API.  So, you never know their actual limit before you hit it. We are searching for a workaround or a web endpoint, but it looks like we will have to test all pairs and hardcode those limits.

**Comment by kroitor** [comment]
@ccasselman Which particular values for $price and $quantity do you use? Your code does not show that, unfortunately...

**Comment by ccasselman** [comment]
Well damn, I had just seen that myself and was scrambling to verify before I posted the solution.

I'm so sorry I posted; however, glad I did because I had been fighting Poloniex `issues` for about a week now and at least I know!

Going to verify and hope to never bother you again.  If you don't hear back that was exactly it!

LOVE the library!  

I am a PHP coder and really wish I knew js better so I could contribute to the project.
